src/garage/torch/algos/vpg.py

```diff
@@ -323,26 +332,37 @@ def process_samples(self, itr, paths):
             pad_to_last(path['rewards'], total_length=self.max_path_length)
             for path in paths
         ])
+        baselines = torch.stack([
+            pad_to_last(self._get_baselines(path),
+                        total_length=self.max_path_length) for path in paths
+        ])
 
-        return valids, obs, actions, rewards
+        return obs, actions, rewards, valids, baselines
 
-    def _log(self, itr, paths, loss_before, loss_after, kl_before, kl,
-             policy_entropy):
+    def evaluate_performance(self, itr, batch):
```

Comment:
please call `RLAlgorithm.evaluate_performance` instead.

Reply:
Our intention for `evaluate_performance()` is not to override the function, but to call it with the batch.

Reply:
How should I incorporate additional metrics, then? Metrics like `LossBefore, KL, Entropy` are not included in `np/algos/base.py: evaluate_performance()`.

Reply:
those can be logged within the algorithm, but the standardized evaluation metrics should always be logged via `RLAlgorithm.evaluate_performance`. This makes sure they are all calculated the same across algorithms.

Reply:
Are you suggesting that I override `evaluate_performance()` and call `RLAlgorithm.evaluate_performance()` in it, plus additional logs.

Reply:
maybe like calling `evaluate_performance()` like
https://github.com/rlworkgroup/garage/blob/master/src/garage/tf/algos/batch_polopt.py#L244
and log your own metrics like
https://github.com/rlworkgroup/garage/blob/master/src/garage/tf/algos/batch_polopt.py#L259

Reply:
Thanks for point this out.